pkg/app/crd_app.go

```diff
@@ -21,20 +21,19 @@ import (
 )
 
 type CRDApp struct {
-	app        *App
-	appModel   *kcv1alpha1.App
-	log        logr.Logger
-	appMetrics *metrics.AppMetrics
-	appClient  kcclient.Interface
+	app          *App
+	appModel     *kcv1alpha1.App
+	log          logr.Logger
+	countMetrics *metrics.ReconcileCountMetrics
+	appClient    kcclient.Interface
 }
 
 // NewCRDApp creates new CRD app
-func NewCRDApp(appModel *kcv1alpha1.App, log logr.Logger, appMetrics *metrics.AppMetrics,
-	appClient kcclient.Interface, fetchFactory fetch.Factory,
+func NewCRDApp(appModel *kcv1alpha1.App, log logr.Logger, appMetrics *metrics.Metrics, appClient kcclient.Interface, fetchFactory fetch.Factory,
 	templateFactory template.Factory, deployFactory deploy.Factory,
 	compInfo ComponentInfo, opts Opts) *CRDApp {
 
-	crdApp := &CRDApp{appModel: appModel, log: log, appMetrics: appMetrics, appClient: appClient}
+	crdApp := &CRDApp{appModel: appModel, log: log, countMetrics: appMetrics.ReconcileCountMetrics, appClient: appClient}
```

Comment:
Don't we need time metrics here? It feels a bit weird to pass the complete metrics object and only use the count metrics.

Reply:
We don't need it in CRDApp so I have just removed it from here. We just need it in `App` object